--- vlc-2.0.5/configure.ac.bero	2013-03-01 00:37:15.622975825 +0100
+++ vlc-2.0.5/configure.ac	2013-03-01 00:37:42.315642120 +0100
@@ -1875,6 +1875,7 @@ dnl
 AC_ARG_ENABLE(smb,
   [AS_HELP_STRING([--disable-smb], [disable SMB/CIFS support (default auto)])])
 if test "${enable_smb}" != "no"; then
+  CPPFLAGS="$CPPFLAGS -I/usr/include/samba-4.0"
   AC_CHECK_HEADERS(libsmbclient.h,
     [ VLC_ADD_PLUGIN([access_smb])
       VLC_ADD_LIBS([access_smb],[-lsmbclient]) ],
