--- vlc-2.0.0/configure.ac.pulse-version	2012-02-17 15:38:25.000000000 +0100
+++ vlc-2.0.0/configure.ac	2012-02-20 11:25:27.624307853 +0100
@@ -3428,7 +3428,7 @@ AC_ARG_ENABLE(pulse,
     [use the PulseAudio client library (default auto)])])
 have_pulse="no"
 AS_IF([test "${enable_pulse}" != "no"], [
-  PKG_CHECK_MODULES([PULSE], [libpulse >= 0.9.22], [
+  PKG_CHECK_MODULES([PULSE], [libpulse >= 0.9.21], [
     have_pulse="yes"
   ], [
     PKG_CHECK_MODULES([PULSE], [libpulse >= 0.9.16], [
--- vlc-2.0.0/modules/audio_output/pulse.c.pulse-version	2011-12-08 19:00:26.000000000 +0100
+++ vlc-2.0.0/modules/audio_output/pulse.c	2012-02-20 11:26:02.990321695 +0100
@@ -33,7 +33,7 @@
 
 #include <pulse/pulseaudio.h>
 #include "vlcpulse.h"
-#if !PA_CHECK_VERSION(0,9,22)
+#if !PA_CHECK_VERSION(0,9,21)
 # include <vlc_xlib.h>
 #endif
 
@@ -664,7 +664,7 @@ static int StreamMove(vlc_object_t *obj,
  */
 static int Open(vlc_object_t *obj)
 {
-#if !PA_CHECK_VERSION(0,9,22)
+#if !PA_CHECK_VERSION(0,9,21)
     if (!vlc_xlib_init(obj))
         return VLC_EGENERIC;
 #endif
