diff --git a/share/Makefile.am b/share/Makefile.am
index 105ed0c..bd3dab0 100644
--- a/share/Makefile.am
+++ b/share/Makefile.am
@@ -79,7 +79,7 @@ skins2/default.vlt: $(skins2_default_vlt_FILES)
 	mkdir -p skins2
 	(cd "$(srcdir)/skins2"; find default -print0 | \
 		LC_ALL=C sort -z | \
-		tar cvv --exclude .svn --no-recursion --null -T -) | \
+		tar cf default.vlt --exclude .svn --no-recursion --null -T -) | \
 	gzip -n > skins2/default.vlt
 
 #
diff --git a/share/Makefile.in b/share/Makefile.in
index 51a77da..71afe6e 100644
--- a/share/Makefile.in
+++ b/share/Makefile.in
@@ -1699,7 +1699,7 @@ skins2/default.vlt: $(skins2_default_vlt_FILES)
 	mkdir -p skins2
 	(cd "$(srcdir)/skins2"; find default -print0 | \
 		LC_ALL=C sort -z | \
-		tar cvv --exclude .svn --no-recursion --null -T -) | \
+		tar cf default.vlt --exclude .svn --no-recursion --null -T -) | \
 	gzip -n > skins2/default.vlt
 
 .lua.luac:
