--- vlc-2.1.1/modules/text_renderer/freetype.c.orig	2013-11-21 19:34:23.543075385 +0100
+++ vlc-2.1.1/modules/text_renderer/freetype.c	2013-11-21 19:37:39.245628766 +0100
@@ -67,10 +67,10 @@
 # define SYSTEM_DEFAULT_MONOSPACE_FONT_FILE "/system/fonts/DroidSansMono.ttf"
 # define SYSTEM_DEFAULT_MONOSPACE_FAMILY "Droid Sans Mono"
 #else
-# define SYSTEM_DEFAULT_FONT_FILE "/usr/share/fonts/truetype/freefont/FreeSerifBold.ttf"
-# define SYSTEM_DEFAULT_FAMILY "Serif Bold"
-# define SYSTEM_DEFAULT_MONOSPACE_FONT_FILE "/usr/share/fonts/truetype/freefont/FreeMono.ttf"
-# define SYSTEM_DEFAULT_MONOSPACE_FAMILY "Monospace"
+# define SYSTEM_DEFAULT_FONT_FILE "/usr/share/fonts/TTF/VeraBd.ttf"
+# define SYSTEM_DEFAULT_FAMILY "Bitstream Vera Bold"
+# define SYSTEM_DEFAULT_MONOSPACE_FONT_FILE "/usr/share/fonts/TTF/liberation/LiberationMono-Regular.ttf"
+# define SYSTEM_DEFAULT_MONOSPACE_FAMILY "Liberation Mono"
 #endif
 
 #ifndef DEFAULT_FONT_FILE
