diff -urN vlc-2.1.2/configure.ac vlc-2.1.2-patched/configure.ac
--- vlc-2.1.2/configure.ac	2013-12-08 03:16:19.000000000 +1100
+++ vlc-2.1.2-patched/configure.ac	2013-12-12 20:26:22.395195372 +1100
@@ -1556,85 +1556,63 @@
 AC_ARG_ENABLE(live555,
   [AS_HELP_STRING([--enable-live555],
     [enable RTSP input through live555 (default enabled)])])
-
 AS_IF([test "${enable_live555}" != "no" -a -n "${CXX}"], [
-  PKG_CHECK_MODULES(LIVE555, live555, [
-    VLC_ADD_PLUGIN([live555])
-    VLC_ADD_CXXFLAGS([live555], [$LIVE555_CFLAGS])
-    VLC_ADD_LIBS([live555],[$LIVE555_LIBS])
+  AC_LANG_PUSH(C++)
+  CPPFLAGS_save="${CPPFLAGS}"
+  AS_IF([test -z "${CONTRIB_DIR}"], [
+    CPPFLAGS_live555="-I${libdir}/live/liveMedia/include -I${libdir}/live/groupsock/include -I${libdir}/live/BasicUsageEnvironment/include -I${libdir}/live/UsageEnvironment/include"
   ], [
-    AC_MSG_WARN([${LIVE555_PKG_ERRORS}.])
-
-    AC_LANG_PUSH(C++)
-    VLC_SAVE_FLAGS
-
-    dnl detect include paths
-    AS_IF([test -f "${CONTRIB_DIR}/include/UsageEnvironment.hh"], [
-      CPPFLAGS_live555="-I${CONTRIB_DIR}/include/liveMedia -I${CONTRIB_DIR}/include/groupsock -I${CONTRIB_DIR}/include/BasicUsageEnvironment -I${CONTRIB_DIR}/include/UsageEnvironment"
-    ], [
-      AS_IF([test "${SYS}" != "os2"], [
-        LIVE555_PREFIX=${LIVE555_PREFIX-"/usr"}
-        CPPFLAGS_live555="-I${LIVE555_PREFIX}/include/liveMedia -I${LIVE555_PREFIX}/include/groupsock -I${LIVE555_PREFIX}/include/BasicUsageEnvironment -I${LIVE555_PREFIX}/include/UsageEnvironment"
-      ], [
-        LIVE555_PREFIX=${LIVE555_PREFIX-"/usr/lib/live"}
-        CPPFLAGS_live555="-I${LIVE555_PREFIX}/liveMedia/include -I${LIVE555_PREFIX}/groupsock/include -I${LIVE555_PREFIX}/BasicUsageEnvironment/include -I${LIVE555_PREFIX}/UsageEnvironment/include"
-        LDFLAGS_live555="-L${LIVE555_PREFIX}/liveMedia -L${LIVE555_PREFIX}/groupsock -L${LIVE555_PREFIX}/BasicUsageEnvironment -L${LIVE555_PREFIX}/UsageEnvironment"
-      ])
-    ])
-
-    dnl CPP Flags
-    AS_IF([test "${SYS}" = "solaris"], [
-      CPPFLAGS_live555="${CPPFLAGS_live555} -DSOLARIS"
-    ])
-    CPPFLAGS="${CPPFLAGS} ${CPPFLAGS_live555}"
-    LDFLAGS="${LDFLAGS} ${LDFLAGS_live555}"
+    CPPFLAGS_live555="-I${CONTRIB_DIR}/include/liveMedia -I${CONTRIB_DIR}/include/groupsock -I${CONTRIB_DIR}/include/BasicUsageEnvironment -I${CONTRIB_DIR}/include/UsageEnvironment"
+  ])
+  AS_IF([test "${SYS}" = "solaris"], [
+    CPPFLAGS_live555="${CPPFLAGS_live555} -DSOLARIS"
+  ])
+  CPPFLAGS="${CPPFLAGS} ${CPPFLAGS_live555}"
 
-    dnl version check
-    AC_CACHE_CHECK([for live555 version 1324598400 or later], [ac_cv_live555], [
-      AC_PREPROC_IFELSE([AC_LANG_PROGRAM([
+  AC_CACHE_CHECK([for live555 version 1324598400 or later], [ac_cv_live555], [
+    AC_PREPROC_IFELSE([AC_LANG_PROGRAM([
 [#include <liveMedia_version.hh>
 #if !defined (LIVEMEDIA_LIBRARY_VERSION_INT) || (LIVEMEDIA_LIBRARY_VERSION_INT < 1324598400)
 # error BOOM
 #endif]])
-      ], [
-        ac_cv_live555="yes"
-      ], [
-        ac_cv_live555="no"
-      ])
+    ], [
+      ac_cv_live555="yes"
+    ], [
+      ac_cv_live555="no"
     ])
-
-    AS_IF([test "$ac_cv_live555" = "no"], [
-      AC_MSG_WARN([live555 is missing or its installed version is too old:
+  ])
+  AS_IF([test "$ac_cv_live555" = "no"], [
+    AC_MSG_WARN([liveMedia is missing or its installed version is too old:
 Version 2011.12.23 or later is required to proceed.
 You can get an updated one from http://www.live555.com/liveMedia .])
-      AS_IF([test -n "${enable_live555}"], [
-        AC_MSG_ERROR([Update live555 or pass --disable-live555 to disable RTSP input support.])
-      ])
-    ], [
-      other_libs="-lgroupsock -lBasicUsageEnvironment -lUsageEnvironment"
-      other_libs_pic="-lgroupsock_pic -lBasicUsageEnvironment_pic -lUsageEnvironment_pic"
-      AS_IF([test "${SYS}" = "mingw32"], [
-        # add ws2_32 for closesocket, select, recv
-        other_libs="$other_libs -lws2_32"
-      ])
+    AS_IF([test -n "${enable_live555}"], [
+      AC_MSG_ERROR([Update live555 or pass --disable-live555 to disable RTSP input support.])
+    ])
+  ], [
+    other_libs="-L${libdir}/live/groupsock -lgroupsock -L${libdir}/live/BasicUsageEnvironment -lBasicUsageEnvironment -L${libdir}/live/UsageEnvironment -lUsageEnvironment"
+    other_libs_pic="-lgroupsock_pic -lBasicUsageEnvironment_pic -lUsageEnvironment_pic"
+    AS_IF([test "${SYS}" = "mingw32"], [
+      # add ws2_32 for closesocket, select, recv
+      other_libs="$other_libs -lws2_32"
+    ], [test "${SYS}" = "mingwce"], [
+      # add ws2 for closesocket, select, recv
+      other_libs="$other_libs -lws2"
+    ])
 
-      dnl We need to check for pic because live555 don't provide shared libs
-      dnl and we want to build a plugins so we need -fPIC on some arch.
-      VLC_ADD_CXXFLAGS([live555], [${CPPFLAGS_live555}])
-      VLC_ADD_LDFLAGS([live555], [${LDFLAGS_live555}])
-      AC_CHECK_LIB(liveMedia_pic, main, [
+    dnl We need to check for pic because live555 don't provide shared libs
+    dnl and we want to build a plugins so we need -fPIC on some arch.
+    VLC_ADD_CXXFLAGS([live555], [${CPPFLAGS_live555}])
+    AC_CHECK_LIB(liveMedia_pic, main, [
+      VLC_ADD_PLUGIN([live555])
+      VLC_ADD_LIBS([live555], [-lliveMedia_pic ${other_libs_pic}])
+    ],[
+      AC_CHECK_LIB(liveMedia, main, [
         VLC_ADD_PLUGIN([live555])
-        VLC_ADD_LIBS([live555], [-lliveMedia_pic ${other_libs_pic}])
-      ],[
-        AC_CHECK_LIB(liveMedia, main, [
-          VLC_ADD_PLUGIN([live555])
-          VLC_ADD_LIBS([live555], [-lliveMedia ${other_libs}])
-        ],[],[${other_libs}])
-      ],[${other_libs_pic}])
-    ])
-    VLC_RESTORE_FLAGS
-    AC_LANG_POP(C++)
+        VLC_ADD_LIBS([live555], [-L${libdir}/live/liveMedia -lliveMedia ${other_libs}])
+      ],[],[-L${libdir}/live/liveMedia ${other_libs}])],[${other_libs_pic}])
   ])
+  CPPFLAGS="${CPPFLAGS_save}"
+  AC_LANG_POP(C++)
 ])
 
 dnl
